### PR TITLE
Initial code for MipInfo ser/der

### PR DESCRIPTION
**Pull request by sydhds**
* [ ] document all added functions
* [ ] try in sandbox /simulation/labnet
* [ ] unit tests on the added/changed features
  * [ ] make tests compile
  * [ ] make tests pass 
* [ ] add logs allowing easy debugging in case the changes caused problems
* [ ] if the API has changed, update the API specification

### CODE REVIEW

File: massa-versioning-worker/src/versioning_ser_der.rs [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+        self.u32_serializer.serialize(&name_len, buffer)?;
+        buffer.extend(value.name.as_bytes());
+        // version
+        self.u32_serializer.serialize(&value.version, buffer)?;
+        // component
+        let component = match &value.component {
+            MipComponent::Address => u32::from(MipComponent::Address),
+            MipComponent::Block => u32::from(MipComponent::Block),
+            MipComponent::VM => u32::from(MipComponent::VM),
+        };
```
Comment (Leo-Besancon) [inline massa-versioning-worker/src/versioning_ser_der.rs:72]: Can't you just do this?
```rust
let component = u32::from(&value.component);
```

File: massa-versioning-worker/src/versioning_ser_der.rs [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+            .serialize(&value.component_version, buffer)?;
+        // start
+        self.time_serializer.serialize(&value.start, buffer)?;
+        // timeout
+        self.time_serializer.serialize(&value.start, buffer)?;
+        Ok(())
+    }
+}
+
+/// Deserializer for VersioningInfo
```
Comment (Leo-Besancon) [inline massa-versioning-worker/src/versioning_ser_der.rs:85]:
```suggestion
/// Deserializer for MipInfo
```

**Review by Leo-Besancon — APPROVED** [review]
Seems good!

File: massa-versioning-worker/src/versioning_ser_der.rs (hunk 1 above)
Comment (sydhds) [reply massa-versioning-worker/src/versioning_ser_der.rs]: fixed

File: massa-versioning-worker/src/versioning_ser_der.rs (hunk 2 above)
Comment (sydhds) [reply massa-versioning-worker/src/versioning_ser_der.rs]: done

**Pull request merged**